examples/replay.cc

```diff
@@ -14,21 +15,23 @@ class Replay : public sc2::ReplayObserver {
         sc2::ReplayObserver() {
     }
 
+    
+
+    //Called once when the game is started (or restarted)
     void OnGameStart() final {
-        const sc2::ObservationInterface* obs = Observation();
-        assert(obs->GetUnitTypeData().size() > 0);
-        count_units_built_.resize(obs->GetUnitTypeData().size());
-        std::fill(count_units_built_.begin(), count_units_built_.end(), 0);
     }
     
+    //Called when a unit is created by the player being observed
     void OnUnitCreated(const sc2::Unit& unit) final {
-        assert(uint32_t(unit.unit_type) < count_units_built_.size());
-        ++count_units_built_[unit.unit_type];
+
     }
 
+    //This is called when stepping through the game, and will not be called if the real time flag is passed to the coordinator.
```

Comment:
I think this still gets call in realtime mode.

Reply:
Also, right now I don't think realtime mode even works for replays?

Reply:
Per the docs "[OnStep] will only get called when stepping. It will not get called in a real time game. In a real time game the user will be responsible for calling GetObservation() via the ObservationInterface."

Reply:
Just tested this. I think the documentation is out of date. It does call OnStep for you, and you also don't need to call GetObservation yourself anymore either.